variable/test/rebin_test.cpp

```diff
@@ -224,3 +225,17 @@ TEST(Variable, rebin_mask_outer_single) {
 
   ASSERT_EQ(result, expected);
 }
+TEST(Variable, check_rebin_cannot_be_used_on_bin_data) {
+  Dimensions dims{Dim::Y, 1};
+  Variable buffer =
+      makeVariable<double>(Dims{Dim::X}, Shape{4}, Values{1, 2, 3, 4});
+  Variable indices = makeVariable<std::pair<scipp::index, scipp::index>>(
+      dims, Values{std::pair{0, 3}});
+  Variable var = make_bins(indices, Dim::X, buffer);
+  const auto oldEdge =
+      makeVariable<double>(Dimensions{Dim::Y, 2}, Values{1, 4});
+  const auto newEdge =
+      makeVariable<double>(Dimensions{Dim::Y, 4}, Values{0, 1, 2, 3});
+  EXPECT_THROW([[maybe_unused]] auto res = rebin(var, Dim::Y, oldEdge, newEdge),
+               except::BucketError);
```

Comment:
I think we should throw a `TypeError` instead? There is no problem with the buckets (bins), it is just a type not accepted by the function.